api/src/controllers/user.js

```diff
@@ -85,6 +85,30 @@ exports.list = (req, res) => {
     }
 }
 
+exports.delete = (req, res) => {
+    // authorize access to the authenticated user model only
+    if (req.params.userId !== req.user.sub) {
+        return res.status(401).send()
+    }
+
+    // check to see if the user exists
+    User.findById(req.params.userId)
```

Comment:
1. Use Async Await
2. Also remove Pins by this user
3. Also remove follows
4. Add a test case for this endpoint

Reply:
I would also recommend pulling from the current users JWT. I think it's `req.user.sub` for the UID.